akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -2129,7 +2131,12 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
     isExpired(key, getUsedTimestamp(key))
 
   def isExpired(key: KeyId, timestamp: Timestamp): Boolean = {
-    expiryEnabled && timestamp != 0L && timestamp <= System.currentTimeMillis() - getExpiryDuration(key).toMillis
+    if (expiryEnabled && timestamp != 0L) {
+      val expiryDuration = getExpiryDuration(key)
+      expiryDuration != Duration.Zero && timestamp <= System.currentTimeMillis() - expiryDuration.toMillis
+    } else {
+      false
+    }
```

Comment:
This was a bug in the gossip exchange when expiry was configured (enabled) but exchanging other keys without expiry. getExpiryDuration is zero for those.